### PR TITLE
Fix Kerberos Authentication failing requests without cookies and excludedPaths config.

### PR DESCRIPTION
**Pull request by nishantmonu51**
KerberosAuthenticator was failing `First` request from the clients.
After authentication we were setting the cookie properly but not
setting the the authenticated flag in the request. This PR fixes that.

Additional Fixes -
1. Excluded paths config - It was not working as expected. Requests for excluded paths were still failing. Fixed this by adding authentication result with anonymous user.
2. Removing of Unused SpnegoFilterConfig - replaced by
KerberosAuthenticator
2. removed unused internalClientKeytab and principal configs from KerberosAuthenticator
3. Fix docs accordingly and add docs for configuring an escalated
client.

Reproducing the Issue - 
- Configure Kerberos Security 
- Perform Kinit on client side and send an HTTP request to any druid endpoint
- Druid returns unauthorized error  with a set-cookie flag. ----- HERE druid should have returned a 200 OK
- Using the cookie on subsequent requests works fine.

### CODE REVIEW

**Comment by nishantmonu51** [comment]
@jon-wei @b-slim Please review.

File: extensions-core/druid-kerberos/src/main/java/io/druid/security/kerberos/KerberosAuthenticator.java [hunk 1]
```diff
@@ -236,6 +230,12 @@ public void doFilter(
 
         String path = ((HttpServletRequest) request).getRequestURI();
         if (isExcluded(path)) {
+          // For excluded paths that does NOT need to be authenticated.
```
Comment (jon-wei) [inline extensions-core/druid-kerberos/src/main/java/io/druid/security/kerberos/KerberosAuthenticator.java]: If there's an authorizer configured + the Kerberos authenticator, this won't be enough to skip security checks on a request (the authorizer would check what permissions the "anonymous" user would have and deny the request unless the user has configured appropriate permissions for the "anonymous" user)

To support `excludedPaths`, there's a similar concept with "unsecured paths" in the various service initialization files (e.g., UNSECURED_PATHS in CoordinatorJettyServerInitializer), one option could be to extend that to allow user-specified whitelist entries (i.e., make `excludedPaths` a general concept applied across all authenticators, and remove `excludedPaths` as a Kerberos-specific config). 

I think `excludedPaths` makes more sense as a system-wide property than an authenticator-specific property, I'll suggest taking this out of this PR and one of us could do follow-on PR to make it a general config.

Comment (nishantmonu51) [reply extensions-core/druid-kerberos/src/main/java/io/druid/security/kerberos/KerberosAuthenticator.java]: @jon-wei: sure, we can make it a general config.
Will raise a separate PR for that.

Comment (nishantmonu51) [reply extensions-core/druid-kerberos/src/main/java/io/druid/security/kerberos/KerberosAuthenticator.java]: @jon-wei, created  https://github.com/druid-io/druid/pull/5614

**Comment by b-slim** [comment]
👍

**Pull request merged**